packages/edit-site/src/components/page-templates/index.js

```diff
@@ -315,6 +323,38 @@ export default function PageTemplates() {
 				elements: authors,
 				width: '1%',
 			},
+			{
+				header: __( 'Post types' ),
+				id: 'postTypes',
+				getValue: ( { item } ) => item.post_types,
+				render: ( { item } ) =>
+					item.post_types
+						?.map(
+							( postType ) => POST_TYPES[ postType ] || postType
+						)
+						?.join( ',' ),
+				elements: [
+					{ value: 'post', label: POST_TYPES.post },
+					{ value: 'page', label: POST_TYPES.page },
+				],
+				filterBy: {
+					operators: [ OPERATOR_IS_ANY ],
+				},
+			},
+			{
+				header: __( 'Type' ),
```

Comment:
> It's odd that the column is either empty or 'custom'. Would it be feasible to add other types? E.g. Archive, Single, Utility? Combined with the post type filter this could potentially work quite well.

To be honest, I'm struggling to see the value of this field/filter in isolation: having the `postTypes` one enables users to search for templates bound that those post types — which are considered custom already, so this `Type` is a redundant. Is there any other value to it I'm not seeing?

Is there any stablished categorization of templates I can look at? That could be useful, though I don't want to introduce a new thing in this PR.

If this field doesn't provide much value right now, I'd rather start with `postTypes` only.

Reply:
I think the main motivator was to provide a way to replicate the 'Custom' section in 6.5's Templates sidebar:

<img width="357" alt="Screenshot 2024-05-29 at 14 49 17" src="https://github.com/WordPress/gutenberg/assets/846565/2f51a13b-0d7d-4640-a4dc-4add4049d98f">

Without this filter it wouldn't be possible to view custom templates easily which could be seen as a regression? Possibly a bad idea, but could the filter be hidden in the UI but still function as part of a 'Custom' view?

---

The [template hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy/#visual-overview) resolves by page type, essentially; "Archive", "Singular", "Front Page", "Posts Page", "Search Results", "404". 

Grouping by "Archive", "Singular", and "Utility" seems reasonable, but I don't think it's defined in the code. Agree it seems late to be adding this complexity.

Reply:
> The [template hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy/#visual-overview) resolves by page type, essentially; "Archive", "Singular", "Front Page", "Posts Page", "Search Results", "404".

I feel this is the best solution, and I'm also wary to introduce it this close to the beta.

If we agree exposing those types would be best long-term, a plan for 6.6 can be:

- naming the field "Type", so we can reuse it later
- its values for now are "Custom" / "Not custom" — we can add more granularity in 6.7 ("Archive", etc.)

Thoughts?

Pushed this approach at [895214a](https://github.com/WordPress/gutenberg/pull/62075/commits/895214a792352a33e1f91d6e713c8d792c41a9e6). Made the field hidden by default and not a badge as well.

Reply:
How do you feel about implementing the filter with `custom` / `not custom` as values, but hiding it in the UI for 6.6?

We can then use it to add a 'Custom' view, including a description about what custom templates are. This would be similar (I think) to categories in the Patterns page.

<img width="868" alt="Screenshot 2024-05-30 at 12 23 21" src="https://github.com/WordPress/gutenberg/assets/846565/ad50aefd-5cc2-4fda-a194-67c486f5e0e9">

This would be dependent on updating the frame titles, but there's a [PR](https://github.com/WordPress/gutenberg/pull/61983) for that ready to go.